pandas/io/parsers.py

```diff
@@ -847,12 +847,18 @@ def _validate_usecols_arg(usecols):
     or strings (column by name). Raises a ValueError
     if that is not the case.
     """
+    msg = ("The elements of 'usecols' must "
+           "either be all strings or all integers")
+
     if usecols is not None:
         usecols_dtype = lib.infer_dtype(usecols)
-        if usecols_dtype not in ('integer', 'string'):
-            raise ValueError(("The elements of 'usecols' "
-                              "must either be all strings "
-                              "or all integers"))
+
+        if usecols_dtype == 'mixed':
```

Comment:
Well I can't comment on the errors that @hassanshamim is getting when running the multibyte, but I was suggesting that there should be a nice helper function to check for a mixed array of `unicode` and `string` (we might be talking about different things here?)